CMakeLists.txt

```diff
@@ -147,7 +147,7 @@ if( TRACCC_SETUP_GOOGLETEST )
 endif()
 
 # option for algebra plugins (ARRAY EIGEN SMATRIX VC VECMEM)
-set(TRACCC_ALGEBRA_PLUGINS EIGEN CACHE STRING "Algebra plugin to use in the build")
+set(TRACCC_ALGEBRA_PLUGINS ARRAY CACHE STRING "Algebra plugin to use in the build")
```

Comment:
This should be OK for now. But It's gonna be tricky when we implement KF which has bunch of matrix operations that doesn't exist in  `ARRAY` algebra-plugins. We need them ASAP regardless of the default plugin though

Reply:
The CUDA code can still be compiled with the EIGEN plugin.

And I intend to make this possible for the SYCL code later on as well. :thinking: That will just need some changes in the CUDA code as well. Because the issue with Eigen at the moment is that with that plugin `traccc::vector2` is not trivially copyable. And as such, it can't be given to a SYCL kernel as an argument. (It's actually interesting that CUDA doesn't complain about this at all. Maybe there's some CUDA specific code in Eigen?)

This should all be possible to work around on our side. But the CUDA code will need to be adjusted as well after we change some of the core types.